plugins/tech-insights/src/api/TechInsightsClient.ts

```diff
@@ -82,6 +82,9 @@ export class TechInsightsClient implements TechInsightsApi {
       {
         method: 'POST',
         body: JSON.stringify(requestBody),
+        headers: {
```

Comment:
wouldn't this be set automatically in the check further down on line 109?

```ts
    if (!request.headers.has('content-type')) {
      request.headers.set('content-type', 'application/json');
```

Reply:
I think this is a case-sensitivity issue. The generic case is all lowercase and the thing added in this PR is Sentence cased. We should probably instead change the generic approach to use upper first letters.

Reply:
I thought the headers object was explicitly case insensitive these days?

Reply:
![Screenshot 2022-10-17 at 17 12 41](https://user-images.githubusercontent.com/3097461/196215279-a75d82af-b7f7-4115-aa3b-3fcb4f0c80a7.png)

Reply:
I don't know why, but at application runtime, when instantiating the `new Request` it automatically loads the `content-type` with `plain/text`

Reply:
This is in the browser console:

![Screenshot 2022-10-18 at 09 22 19](https://user-images.githubusercontent.com/3097461/196363457-c4f5b019-10b5-4432-8430-c326ffd5b006.png)

This is in Node:

![Screenshot 2022-10-18 at 09 24 48](https://user-images.githubusercontent.com/3097461/196364028-957b6552-dd38-41eb-8a65-4519b60db6d0.png)

And I tried with all types of casing on those. That content type change is really not something it should do ... 🤔 I wonder what's really up here.

Reply:
Right, it was the scenario that @Irma12 just talked about, at some point this will return us `text/plain; charset=UTF-8`, if you prefer, I can demonstrate the error happening

Reply:
I would have liked to see a test written that you initially see is failing, and which then is fixed by the client change. We often use `swc` to create fake servers to talk to in tests.

Reply:
Yes, I can do this, do you have an example?

- Image with the scenario happening below
![image](https://user-images.githubusercontent.com/56898864/196380569-6042db4e-a208-484d-a5a6-ecaa65629a91.png)

Reply:
Aha yeah I see, this is what it did _before_ the changes. Thanks for the screenshot. That makes sense; as a security matter the request constructor avoids leaving the content type empty or wrong. So a safer fix might have been to read out of the `init` parameter to the function, instead of out of the copy which has a complex behavior that we explicitly don't want to rely on here.

Reply:
Regarding msw tests (i messed up and referred to the wrong package name ...) - you can search the repo for `from 'msw'`, or check out for example [this one](https://github.com/backstage/backstage/blob/13c32d12e9aa9b55d5a51d17c169c6fda9f36601/packages/backend-common/src/reading/FetchUrlReader.test.ts#L29).